### PR TITLE
ARUHA-293: switch from configuration to autowired

### PR DESCRIPTION
**Pull request by v1ctor**

### CODE REVIEW

File: src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java [hunk 1]
```diff
@@ -59,6 +60,7 @@
     private final MetricRegistry metricRegistry;
     private final ClosedConnectionsCrutch closedConnectionsCrutch;
 
+    @Autowired
     public EventStreamController(final EventTypeRepository eventTypeRepository, final TopicRepository topicRepository,
```
Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java]: Which class is being injected in the first parameter?

Comment (v1ctor) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java]: \@Primary but for readability I could do it \@Cache

Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java]: I think it's fine. You already made it explicit when a `@DB` is required, so it's all fine 👍

File: src/main/java/de/zalando/aruha/nakadi/annotations/DB.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.FIELD,
+        ElementType.METHOD,
+        ElementType.TYPE,
+        ElementType.PARAMETER})
+@Retention(RetentionPolicy.RUNTIME)
+@Qualifier
+public @interface DB {
```
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/annotations/DB.java]: What exactly this annotation does? I think I don't get it.

Comment (v1ctor) [inline src/main/java/de/zalando/aruha/nakadi/annotations/DB.java]: I specify to inject particular EventTypeRepository (EventTypeDBRepository) in CachingEventTypeRepository

**Comment by adyach** [comment]
For me it is fine 👍

**Comment by rcillo** [comment]
:shipit:

**Comment by adyach** [comment]
:shipit:

**Pull request merged**